spec/index.bs

```diff
@@ -8263,8 +8263,8 @@ enum GPUBlendOperation {
 dictionary GPUDepthStencilState {
     required GPUTextureFormat format;
 
-    boolean depthWriteEnabled = false;
-    GPUCompareFunction depthCompare = "always";
+    required boolean depthWriteEnabled;
+    required GPUCompareFunction depthCompare;
```

Comment:
Hm... arguably for the same reason, maybe these should be required only if the format has a depth aspect?

Reply:
Though like with the other case it's annoying for implementations because now this boolean becomes tri-state.

Reply:
That wouldn't be a problem if we made it an exception instead of a validation rule. But that would be a weird carveout.

Reply:
> Hm... arguably for the same reason, maybe these should be required only if the format has a depth aspect?

Yeah, sounds like it is more consistent although the implementation could be more complex as depthClearValue... If no objection, I will update the PR soon.

> That wouldn't be a problem if we made it an exception instead of a validation rule.

Would you mind explaining it a bit more for me please because I couldn't get well? (Honestly I haven't comprehend the difference exception and validation rule well yet.)

Reply:
If the error is raised as an exception then it can be implemented at the JavaScript bindings level (ie: Blink in Chromium), which is quite a bit simpler. If it's done as a validation error it needs to be passed all the way through the WebGPU native implementation, which then needs to figure out things like how to handle a tri-state boolean in a C API.

The reasons for preferring a validation error are that it's more consistent with the handling of every other error that is raised from this part of the code, and there may be some cases (not here) where some of the information necessary to validate are only tracked in the WebGPU implementation.

Reply:
Thanks for the explanations!